### ISSUE TITLE
Real-time file monitoring stops working if scan removed file.

### ISSUE THREAD

**Issue by super-a1ice** [wazuh/wazuh#1034]
## Description
1. Deploy Wazuh agent (v3.4.0-1) on Windows 10 Pro
2. Add real-time monitoring for C:\User directory in agent ossec.conf
```
<directories check_all="no" realtime="yes">C:\Users</directories>
```
3. Check that C:\Users directory is being monitored in real time by wazuh agent and wait until real-time file monitoring starts
agent ossec.log

```
2018/08/01 01:38:21 ossec-agent: INFO: Initializing real time file monitoring engine.
2018/08/01 01:38:21 ossec-agent: INFO: Active response command not present: 'active-response/bin/restart-ossec.sh'. Not using it on this system.
2018/08/01 01:38:21 ossec-agent: INFO: Real time file monitoring engine started.
2018/08/01 01:38:21 ossec-agent: INFO: Finished creating syscheck database (pre-scan completed).
2018/08/01 01:38:26 ossec-agent: INFO: Ending syscheck scan (forwarding database).
2018/08/01 01:38:36 ossec-agent: INFO: Starting syscheck real-time monitoring.
```

4. Open google chrome and download some file.

5. This error appears in agent ossec.log file and after that real-time file monitoring stops working.

```
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\1VIM5T8MVXCE6LQLJON8.temp': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\1VIM5T8MVXCE6LQLJON8.temp': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\1VIM5T8MVXCE6LQLJON8.temp': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\969252ce11249fdd.customDestinations-ms~RF1af1e813.TMP': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\969252ce11249fdd.customDestinations-ms~RF1af1e813.TMP': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\969252ce11249fdd.customDestinations-ms~RF1af1e813.TMP': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\1VIM5T8MVXCE6LQLJON8.temp': No such file or directory 
2018/08/01 01:38:37 ossec-agent: WARNING: Error opening directory: 'C:\Users\john\AppData\Roaming\Microsoft\Windows\Recent\CustomDestinations\969252ce11249fdd.customDestinations-ms~RF1af1e813.TMP': No such file or directory 
2018/08/01 01:38:54 ossec-agent: ERROR: real time call back called, but 0 bytes.
```
## Related issue
ossec-hids issue ![#1402](https://github.com/ossec/ossec-hids/issues/1402)

**Comment by TJOSERAFAEL** [comment]
Hi @super-a1ice,

thank you for reporting this issue. We will fix this problem ASAP. I will let you know when the fix is ready.

Best regards

**Comment by vikman90** [comment]
Hi @super-a1ice,

Thank you very much for reporting this bug. Indeed, if the internal buffer gets full, Windows reports an error condition, in that case, the folder watcher was not refreshed and this produced this issue, just like in OSSEC.

We managed to reproduce it with your configuration and browsing some webs with Internet Explorer. This made us discover another bug: if the agent failed to get the owner of a file (due to missing permissions or that the file was just deleted) the agent may crash.

Both bugs have been fixed just now in the branch 3.5 for the upcoming version (PR 
https://github.com/wazuh/wazuh/pull/1062). Hope you find it useful.

Thank you again.
Best regards.
